src/document.c

```diff
@@ -825,7 +825,8 @@ GeanyDocument *document_new_file_if_non_open(void)
 }
 
 
-/**
+/** @girskip
```

Comment:
BTW, what is the real issue with `document_new_file()`?  From the API POV it IS a constructor, and in practice it mostly actually is.  The only difference is that it doesn't give ownership to the caller, but anyway the only way to destruct a document is to close it, so I don't really see the problem.
Admittedly I didn't test that with GI, but I fail to see the issue, and the current name makes more sense to me, so I'd rather not change it without being forced and convinced.

With more and more and more stuff changing and added to please g-ir-scanner, I'm starting to think maybe a maintained compatibility layer specifically for it would make more sense, leaving current API alone.  Sure, some changes for it were real nice for everyone, but some look arbitrary and are just a pain to port to with no benefit and no justification for the non-GI user.